CRM/Member/Form/Membership.php

```diff
@@ -122,16 +122,17 @@ class CRM_Member_Form_Membership extends CRM_Member_Form {
   public static function getSelectedMemberships($priceSet, $params) {
     $memTypeSelected = array();
     $priceFieldIDS = self::getPriceFieldIDs($params, $priceSet);
-    if (isset($params['membership_type_id']) && !empty($params['membership_type_id'][1])) {
-      $memTypeSelected = array($params['membership_type_id'][1] => $params['membership_type_id'][1]);
-    }
-    else {
-      foreach ($priceFieldIDS as $priceFieldId) {
-        if ($id = CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceFieldValue', $priceFieldId, 'membership_type_id')) {
-          $memTypeSelected[$id] = $id;
-        }
+
+    foreach ($priceFieldIDS as $priceFieldId) {
+      if ($memTypeID = CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceFieldValue', $priceFieldId, 'membership_type_id')) {
+        $memTypeSelected[$priceFieldId] = $memTypeID;
       }
     }
+
+    if (empty($memTypeSelected) && isset($params['membership_type_id']) && !empty($params['membership_type_id'][1])) {
```

Comment:
I don't like the 1's here as a coding standard matter. Could we do something better?